scripts/r.reclass.area/r.reclass.area.py

```diff
@@ -163,18 +168,29 @@ def reclass(inf, outf, lim, clump, diag, les):
     p1.wait()
     p2.stdin.close()
     p2.wait()
+    expression = "$outfile = $recfile"
+    mapcalc_kwargs = {"outfile": outfile, "recfile": recfile}
     if p2.returncode != 0:
         if lesser:
-            grass.fatal(
-                _("No areas of size less than or equal to %f " "hectares found.")
-                % limit
+            op_str = "less"
+        else:
+            op_str = "greater"
+        if nodata is True:
+            grass.warning(
+                _(
+                    "No areas of size %s than or equal to %f "
+                    "hectares found. Creating NULL raster..."
+                )
+                % (op_str, limit)
             )
+            expression = "$outfile = null()"
+            del mapcalc_kwargs["recfile"]
         else:
             grass.fatal(
-                _("No areas of size greater than or equal to %f " "hectares found.")
-                % limit
+                _("No areas of size %s than or equal to %f " "hectares found.")
+                % (op_str, limit)
```

Comment:
To improve the code overall, please use `.format()` when rewriting the messages.